server/backends/blobstore/util/util.go

```diff
@@ -121,3 +127,82 @@ func RecordDeleteMetrics(typeLabel string, startTime time.Time, err error) {
 		metrics.BlobstoreTypeLabel: typeLabel,
 	}).Observe(float64(duration.Microseconds()))
 }
+
+type compressedWriter struct {
+	ctx       context.Context
+	committer interfaces.Committer
+	zw        io.WriteCloser
+	closer    io.Closer
+	label     string
+	n         int
+}
+
+func NewCompressedBlobStoreWriter(
+	ctx context.Context,
+	committer interfaces.Committer,
+	writer io.Writer,
+	closer io.Closer,
+	label string,
+) interfaces.CommittedWriteCloser {
+	return &compressedWriter{
+		ctx: ctx, committer: committer,
+		zw:     newCompressWriter(writer),
+		closer: closer,
+		label:  label,
+		n:      0,
+	}
+}
+
+func (c *compressedWriter) Commit() error {
+	c.zw.Close()
```

Comment:
if `c.zw.Close()` fails, should we assume something went wrong and abort - i.e. return the error instead of calling `committer.Commit()` ?

Reply:
Good point, done.